### ISSUE TITLE
how can I set the width and height of the panel (32x32)?

### ISSUE THREAD

**Issue by polygontwist** [VGottselig/ESP32-RGB-Matrix-Display#7]
I found this in the ESP32RGBmatrixPanel.h:

```
#define COLUMNS 64
#define ROWS 32

```
How can I overwrite this before initializing (ESP32RGBmatrixPanel matrix;) in my script (for 32x32)?
Or can I only set this in the ESP32RGBmatrixPanel.h?

**Comment by VGottselig** [comment]
Change it to 
#define COLUMNS 32
#define ROWS 32
?

**Comment by polygontwist** [comment]
yes, I didn't want to set it directly in the ESP32RGBmatrixPanel.h. In the *.ino file this did not work after the include. I didn't want to change your code to keep it compatible with other displays.

**Comment by VGottselig** [comment]
You have to change constructors and add parameters rows and columns to it.
And may be  colorChanellsCount too.
Some modules have R1, G1, B1, R2, G2, B2 => colorChanellsCount = 2.
Some modules have just R1, G1, B1=> colorChanellsCount = 1.

**Comment by skraus86** [comment]
I also have this issue, but I'm pretty sure the problem is the constructor. The 64x32 and 32x32 panels are a different scan rate (1:8) versus 16x32 (1:4) so we'd need to re-write the constructor to support it.

I'm not sure how to properly re-write it, but here's an example Arduino project for driving 1:4 panels:

https://forum.arduino.cc/index.php?topic=503416.0
